docs/packages/cluster/AUTOSCALING.md

```diff
@@ -0,0 +1,138 @@
+
+## Why memory?
+
+The autoscaler uses CPU usage for GG nodes and memory usage for job nodes. This is because certain cloud providers like linode do not provide an actual value for the speed of the CPU, but rather the amount of cores. This is problematic because we use Nomad's API for determining the usage on any given node, and it returns its stats in MHz.
+
+## Hardware failover
+
+Before a job server provisioned, we don't know for sure what its specs will be because of the hardware failover system in `cluster-server-provision`. In the autoscaling process, all servers that aren't provisioned yet are assumed to have the specs of the first hardware option in the list.
+
+### Failover has lower specs
+
+In the event that the hardware which ended up being provisioned has lower specs than the first hardware in the list, the autoscaler will calculate the error between how much was expected and how much was actually provisioned. This error number corresponds to how many more servers might be needed to reach the desired server count.
+
+Here is an example of the process in action:
+
+| time since start | desired count | expected total memory | actual total memory |
+| ---------------- | ------------- | --------------------- | ------------------- |
+| 0s               | 2             | 2000MB                | 0MB                 |
+
+We start with 0 servers provisioned, and 2 desired. Our config consists of two hardwares, the first having 1000MB of memory and the second having 500MB of memory. With our failover system if the first one fails to provision, the second will be provisioned.
+
+| time since start | desired count | expected total memory | actual total memory |
+| ---------------- | ------------- | --------------------- | ------------------- |
+| 0s               | 2             | 2000MB                | 0MB                 |
+| 15s              | 3             | 2000MB                | 1000MB              |
+
+After the first iteration, the autoscaler provisioned 2 servers which both ended up failing over and only providing a total of 1000MB of memory. The autoscaler then proceeds to calculate the error like so:
+
+```rust
+ceil(expected - actual) / expected_memory_per_server)
+
+ceil((2000 - 1000) / 1000) = 1
+```
+
+So an extra server was added to the desired count.
+
+Now, if the next server to be provisioned ends up having 1000MB like it should, we will end up having the original amount of desired memory.
+
+| time since start | desired count | expected total memory | actual total memory |
+| ---------------- | ------------- | --------------------- | ------------------- |
+| 0s               | 2             | 2000MB                | 0MB                 |
+| 15s              | 3             | 2000MB                | 1000MB              |
+| 30s              | 3             | 2000MB                | 2000MB              |
+
+The error calculation would now be:
+
+```rust
+ceil((3000 - 2000) / 1000) = 1
```

Comment:
This might be a bit verbose, but I think I would benefit more from having the steps written out:
```suggestion
(3000 - 2000) / 1000 = 1
ceil(1) = 1
```

And below would be
```
(3000 - 1500) / 1000 = 1.5
ceil(1.5) = 2
```

Reply:
The only thing I see this adding is explaining how ceil works, the rest is just algebra. I don't think its necessary to show a step for something you can just easily look up